### PR TITLE
Update lucas_series.py to include another method

### PR DESCRIPTION
**Pull request by Kush1101**
Added another method to calculate lucas_numbers

### **Describe your change:**

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by 53jk1 — APPROVED** [review]

File: maths/lucas_series.py [hunk 1]
```diff
@@ -1,22 +1,64 @@
...
-    print("Lucas series:")
-    for i in range(limit):
-        print(recur_luc(i))
+    from doctest import testmod
+
+    testmod()
+    n = int(input("Enter the number of terms in lucas series:\n").strip())
+    print("Using recursive function to calculate lucas series:")
+    for i in range(n):
+        print(recursive_lucas_number(i), end=" ")
```
Comment (cclauss) [inline maths/lucas_series.py:61]: Please do this on one line with `" ".join(<generator expression>)`

File: maths/lucas_series.py [hunk 2]
```diff
@@ -1,22 +1,64 @@
...
+    print("\nUsing dynamic function to calculate lucas series:")
+    for i in range(n):
+        print(dynamic_lucas_number(i), end=" ")
```
Comment (cclauss) [inline maths/lucas_series.py:64]: Please do this on one line with `" ".join(<generator expression>)`

File: maths/lucas_series.py [hunk 3]
```diff
@@ -1,22 +1,64 @@
-# Lucas Sequence Using Recursion
-
-
-def recur_luc(n):
+def recursive_lucas_number(n):
```
Comment (cclauss) [inline maths/lucas_series.py:1]: How about a URL to a source that discusses this algorithm?

**Review by poyea — APPROVED** [review]

**Comment by Kush1101** [comment]
@cclauss, should we merge it?

File: maths/lucas_series.py [hunk 4]
```diff
@@ -1,22 +1,66 @@
...
```
Comment (shellhub) [inline maths/lucas_series.py:1]:
```suggestion
"""
https://en.wikipedia.org/wiki/Lucas_number
"""


def recursive_lucas_number(n: int) -> int:
```

File: maths/lucas_series.py [hunk 5]
```diff
@@ -1,22 +1,66 @@
...
     """
-    >>> recur_luc(1)
+    https://en.wikipedia.org/wiki/Lucas_number
```
Comment (shellhub) [inline maths/lucas_series.py:3]:
```suggestion
```

File: maths/lucas_series.py [hunk 6]
```diff
@@ -1,22 +1,66 @@
...
-    return recur_luc(n - 1) + recur_luc(n - 2)
+    if not isinstance(n, int):
+        raise TypeError("recursive_lucas_number accepts only integer arguments.")
+
+    return recursive_lucas_number(n - 1) + recursive_lucas_number(n - 2)
+
+
+def dynamic_lucas_number(n: int) -> int:
+    """
+    https://en.wikipedia.org/wiki/Lucas_number
```
Comment (shellhub) [inline maths/lucas_series.py:30]:
```suggestion
```

**Pull request merged**